homeassistant/components/generic_thermostat/climate.py

```diff
@@ -325,7 +325,7 @@ async def async_set_hvac_mode(self, hvac_mode):
             _LOGGER.error("Unrecognized hvac mode: %s", hvac_mode)
             return
         # Ensure we update the current operation after changing the mode
-        self.schedule_update_ha_state()
+        self.async_schedule_update_ha_state()
```

Comment:
Can this use `self.async_write_ha_state()` instead?

Reply:
Yes it should.
```suggestion
        self.async_write_ha_state()
```